fix: Don't throw for unknown network option

spypsy opened the pull request:
We are currently failing on `next-net` deployments with:
```
Failed to fetch network config for network: next-net
```

When a network that's not part of our config is used, we shouldn't throw if the fetch is successful. Instead we just don't hydrate any env and keep going.

alexghr reviewed (commented):
~TBH I think this code should only crash the node if fail to pull the config file from github/the fallback. It shouldn't crash because the network name is not in the json file.~ my bad

spypsy committed:
fix: Don't throw for unknown network option

We are currently failing on `next-net` deployments with:
```
Failed to fetch network config for network: next-net
```

When a network that's not part of our config is used, we shouldn't throw if the fetch is successful. Instead we just don't hydrate any env and keep going.
